ruleset/rules/0770-gitlab_rules.xml

```diff
@@ -1,249 +1,181 @@
-
-{"method":"PUSH","path":"/gitlab/gitlab-ce/issues/1234","format":"html","controller":"Projects::IssuesController","action":"show","status":400,"duration":229.03,"view":174.07,"db":13.24,"time":"2017-08-08T20:15:54.821Z","params":[{"key":"param_key","value":"param_value"}],"remote_ip":"18.245.0.1","user_id":1,"username":"admin","gitaly_calls":76,"gitaly_duration":7.41,"queue_duration": 112.47}
-
-{"method":"PUSH","path":"/gitlab/gitlab-ce/issues/1234","format":"html","controller":"Projects::IssuesController","action":"show","status":300,"duration":229.03,"view":174.07,"db":13.24,"time":"2017-08-08T20:15:54.821Z","params":[{"key":"param_key","value":"param_value"}],"remote_ip":"18.245.0.1","user_id":1,"username":"admin","gitaly_calls":76,"gitaly_duration":7.41,"queue_duration": 112.47}
-
-api_json
-
-{"time":"2018-10-29T12:49:42.123Z","severity":"INFO","duration":709.08,"db":14.59,"view":694.49,"status":200,"method":"GET","path":"/api/v4/projects","params":[{"key":"action","value":"git-upload-pack"},{"key":"changes","value":"_any"},{"key":"key_id","value":"secret"},{"key":"secret_token","value":"[FILTERED]"}],"host":"localhost","ip":"::1","ua":"Ruby","route":"/api/:version/projects","user_id":1,"username":"root","queue_duration":100.31,"gitaly_calls":30,"gitaly_duration":5.36}
-
-
+  Rules for:
+    GitLab ID:  65600 - 65699
 -->
 
-<group name="gitlab_v.12_production_json">
-    <rule id="65600" level="3">
-        <decoded_as>json</decoded_as>
-        <regex>"method":"\w+","path":"/gitlab/gitlab-ce/\w+/\.+","format":"\w+","controller":"\.+","action":"\w+"</regex>
-        <match>"status":200</match>
-        <options>no_full_log</options>
-        <description>(Gitlab)$(method) request Completed Succesfully</description>
-    </rule>
-    <rule id="65602" level="5">
-        <decoded_as>json</decoded_as>
-        <regex>"method":"\w+","path":"/gitlab/gitlab-ce/\w+/\.+","format":"\w+","controller":"\.+","action":"\w+"</regex>
-        <match>"status":400</match>
-        <options>no_full_log</options>
-        <description>(Gitlab)ERROR: couldn't complete $(method) request</description>
-    </rule>
-    <rule id="65603" level="5">
-        <decoded_as>json</decoded_as>
-        <regex>"method":"\w+","path":"/gitlab/gitlab-ce/\w+/\.+","format":"\w+","controller":"\.+","action":"\w+"</regex>
-        <match>"status":300</match>
-        <options>no_full_log</options>
-        <description>(Gitlab)REDIRECTION:The $(method) request has more than one possible response</description>
-    </rule>
-</group>
-<group name="gitlab_v.12_api_json">
-<rule id="65623" level="3">
+<group name="gitlab_v.12">
+    
+  <rule id="65600" level="3">
     <decoded_as>json</decoded_as>
-    <regex>"method":"\w+","path":"/api/\w+/\.+","params":\.+</regex>
+    <regex>"method":"\w+","path":"/gitlab/gitlab-ce/\w+/\.+","format":"\w+","controller":"\.+","action":"\w+"</regex>
```

Comment:
Since fields are already decoded it would be better to use `field` directly to look for the field content we need. Replacing:
```xml
<regex>"method":"\w+","path":"/gitlab/gitlab-ce/\w+/\.+","format":"\w+","controller":"\.+","action":"\w+"</regex>
<match>"status":200</match>
```
by
```xml
<field name="path">^/gitlab/gitlab-ce/</field>
<field name="status">^200$</field>
```

This applies to all rules in this file using `regex` and decoded by the JSON decoder.